plugins/processors/dcos_metadata/dcos_metadata.go

```diff
@@ -240,22 +247,24 @@ func (dm *DCOSMetadata) getClient() (*httpcli.Client, error) {
 	return client, nil
 }
 
-// getContainerID retrieves the container ID linked to this task. Task can have
-// multiple statuses. Each status can have multiple container IDs. In DC/OS,
-// there is a one-to-one mapping between tasks and containers; however it is
-// possible to have nested containers. Therefore we use the first status, and
-// return its parent container ID if possible, and if not, then its ID.
-func getContainerID(statuses []mesos.TaskStatus) string {
+// getContainerIDs retrieves the container ID and the parent container ID of a
+// task from its TaskStatus. The container ID corresponds to the task's
+// container, the parent container ID corresponds to the task's executor's
+// container.
+func getContainerIDs(statuses []mesos.TaskStatus) (string, string) {
```

Comment:
when `pcid` could be empty?

I like named return variable. It's got quick info what is returned. `string, string` means nothing for me. I'd prefer to alias it with something more meaningful.
```suggestion
func getContainerIDs(statuses []mesos.TaskStatus) (containerID string, parentContainerID string) {
```